nova/virt/vmwareapi/vmops.py

```diff
@@ -3172,15 +3176,60 @@ def _sync_sync_server_group(context, sg_uuid):
                 LOG.debug('Sync for server-group %s done', sg_uuid)
                 return
 
+            # First we check for all instances, which have ongoing migrations
+            # on the given host, either as source or destination
+
+            # Decision matrix for migrations:
+            # Mig-Status Action/Host
+            #            Source  Dest
+            # Preparing  Remove  N/A
+            # Running    Remove  Add
+            # (Other states are consistent with default behaviour)
+            #
+            # So the Instance.host will always be the source of the migration,
+            # and we want to remove the rules.
+            # We only need to handle specially the case a running migration
+            # on the destination host, and add it
+
+            MigrationList = objects.migration.MigrationList
+            filters = {
+                "host": host,
+                "instance_uuid": sg.members,
+                "status": ["preparing", "running"],
```

Comment:
Do we automatically exclude deleted Migrations here?

Reply:
Deleted migrations in the state preparing or running? I can add that constraint, I was assuming they are either in the state completed or error.

Reply:
Checking the code, it seems you have to pass `read_deleted="yes"` to the model-query, otherwise it will filter deleted rows automatically